src/txn/txn_ckpt.c

```diff
@@ -95,6 +95,11 @@ __checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
     conn->flush_ckpt_complete = false;
     conn->flush_most_recent = conn->ckpt_most_recent;
     conn->flush_ts = conn->txn_global.last_ckpt_timestamp;
+    /*
+     * It would be more efficient to return here if no tiered storage is enabled in the system.
+     * If the user asks for a flush_tier without tiered storage, the loop below is effectively a
+     * no-op and will not be incorrect. But we could also just return.
```

Comment:
Good point. The metadata is sorted by URI, isn't it? Looking at the following loop I wonder if instead of walking every metadata entry we could search for the first `tiered:` entry and then iterate from there until we hit something that doesn't match. This wouldn't quite be a no-op without tiered storage, but it would be much quicker.

Reply:
Sure. That is way beyond the scope of this ticket whose intent was just documentation.

Reply:
Yup. Understood. I just couldn't resist brainstorming. :-) Not expecting a change on this ticket.